playground/app.js

```diff
@@ -28,6 +28,41 @@ const cmOptions = {
   tabSize: 2,
 };
 
+class GeoPosition extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {...props.formData};
+  }
+
+  onChange(name) {
+    return (event) => {
+      this.setState({
+        [name]: parseFloat(event.target.value)
+      }, () => this.props.onChange(this.state));
```

Comment:
Oh, ok, I think I got it. We first set the state and then we call `props.onChange`, probably to propagate the action.

I was mistaken by:

- The `setState` which isn't promise-based.
- The fact that we're calling `onChange` on the props, probably to cascade the changes.